prdoc/pr_7286.prdoc

```diff
@@ -0,0 +1,29 @@
+title: Remove feature flag checks for Elastic Scaling MVP
+doc:
+  - audience: Runtime Dev
+    description: |
+      ## Description
+      This PR addresses [issue #6352](https://github.com/paritytech/polkadot-sdk/issues/6352),
+      which proposes cleaning up the codebase by assuming that the FeatureIndex::ElasticScalingMVP
+      feature is always enabled.
+
+      ## Integration
```

Comment:
Prdocs are not a copy of the pr description, they should be a summary of the changes and how/if that impacts others